src/main/main.c

```diff
@@ -234,7 +234,7 @@ void init(void)
     servoMixerInit(masterConfig.customServoMixer);
 #endif
 
-#ifdef SPEKTRUM_BIND
+#if defined(SERIAL_RX) && defined(SPEKTRUM_BIND)
```

Comment:
In a way, considering how the build is set up at the moment, without an extra layer of

```
#if defined(X) && defined(Y)
#define USE_y
#endif
```

(after all of the config has been evaluated), I would almost blame the error on the `#undef`, and see the best fix as adding an `#undef SPEKTRUM_BIND` to wherever there is an `#undef SERIAL_RX`.

By far the most common use case is features being enabled by `#define X`, `#undef Y` is a rare edge case, and probably a good candidate for a condition reversal to Y not being defined by default, but instead only by targets that support it.

I would postulate that the build conditions should be treated in a way so that all possible meaningful permutations of conditions should result in buildable configurations, whereas, when `#undef` is being used, it should be up to the author to make sure all potential dependent conditions are resolved with appropriate `#undef`s as well. @ledvinap, @martinbudden, your thoughts?